src/app/pages/storage2/components/dashboard-pool/export-disconnect-modal/export-disconnect-modal.component.ts

```diff
@@ -0,0 +1,244 @@
+import { PoolAttachment } from 'app/interfaces/pool-attachment.interface';
+import { Pool } from 'app/interfaces/pool.interface';
+import { Process } from 'app/interfaces/process.interface';
+import { SystemDatasetConfig } from 'app/interfaces/system-dataset-config.interface';
+import { EntityJobComponent } from 'app/modules/entity/entity-job/entity-job.component';
+import IxValidatorsService from 'app/modules/ix-forms/services/ix-validators.service';
+import { AppLoaderService, DialogService, WebSocketService } from 'app/services';
+
+@UntilDestroy()
+@Component({
+  styleUrls: ['./export-disconnect-modal.component.scss'],
+  templateUrl: './export-disconnect-modal.component.html',
+})
+export class ExportDisconnectModalComponent implements OnInit {
+  readonly helptext = helptext;
+
+  readonly nameInputRequired = this.validatorsService.withMessage(
+    Validators.required,
+    this.translate.instant('Name of the pool is required'),
+  );
+
+  readonly nameInputMustMatch = this.validatorsService.withMessage(
+    Validators.pattern(new RegExp(`^${this.pool.name}$`)),
+    this.translate.instant('Name of the pool must be correct'),
+  );
+
+  showSysDatasetWarning: boolean;
+  showPoolDetachWarning: boolean;
+  showUnknownStatusDetachWarning: boolean;
+  showDestroy: boolean;
+
+  confirmLabelText = this.translate.instant(helptext.exportDialog.confirm);
+  process = {
+    knownProcesses: [] as Process[],
+    unknownProcesses: [] as Process[],
+  };
+
+  attachments: PoolAttachment[] = [];
+  processes: Process[] = [];
+  systemConfig: SystemDatasetConfig;
+
+  isFormLoading = false;
+  form = this.fb.group({
+    destroy: [false],
+    cascade: [true],
+    confirm: [false, [Validators.requiredTrue]],
+    nameInput: ['', [
+      this.nameInputRequired,
+      this.validatorsService.validateOnCondition(
+        (control: AbstractControl) => control.parent?.get('destroy').value,
+        this.nameInputMustMatch,
+      ),
+    ]],
+  });
+
+  restartServices = false;
+
+  constructor(
+    private fb: FormBuilder,
+    private dialogRef: MatDialogRef<ExportDisconnectModalComponent>,
+    private translate: TranslateService,
+    private validatorsService: IxValidatorsService,
+    private dialogService: DialogService,
+    private matDialog: MatDialog,
+    private loader: AppLoaderService,
+    private ws: WebSocketService,
+    @Inject(MAT_DIALOG_DATA) public pool: Pool,
+  ) {}
+
+  ngOnInit(): void {
+    if (this.pool.status === PoolStatus.Unknown) {
+      this.prepareForm();
+      return;
+    }
+
+    this.loadRelatedEntities();
+  }
+
+  cancel(): void {
+    this.dialogRef.close(false);
+  }
+
+  // TODO: change to reactive forms
+  onDestroyCheckedStateChanged($event: unknown): void {
+    if (!$event) {
+      this.resetNameInputValidState();
+    }
+  }
+
+  // TODO: Reactive
+  onConfirmCheckedStateChanged(): void {
+    if (!this.form.get('destroy').value) {
+      this.resetNameInputValidState();
+    }
+  }
```

Comment:
Please use valueChange pipe on init instead of model change event.
```
    this.form.controls.destroy.valueChanges.pipe(untilDestroyed(this)).subscribe((destroy) => {
      if (!destroy) {
        this.form.get('nameInput').reset();
        this.form.get('nameInput').setErrors(null);
      }
    });
```